krabs/krabs/ut.hpp

```diff
@@ -113,19 +122,20 @@ namespace krabs { namespace details {
         // TODO: Only forward the calls that are requested to each provider.
         for (auto &provider : trace.providers_) {
             if (provider_flags.find(provider.get().guid_) != provider_flags.end()) {
-                provider_flags[provider.get().guid_].flags_tuple_ = std::make_tuple<UCHAR, ULONGLONG, ULONGLONG, UCHAR> (0, 0, 0, 0);
+                provider_flags[provider.get().guid_].filter_flags_ = {};
             }
 
-            std::get<0>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().level_;
-            std::get<1>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().any_;
-            std::get<2>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().all_;
-            std::get<3>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().trace_flags_;
+            auto& flags = provider_flags[provider.get().guid_];
+            flags.filter_flags_.level_ |= provider.get().level_;
+            flags.filter_flags_.any_ |= provider.get().any_;
+            flags.filter_flags_.all_ |= provider.get().all_;
+            flags.filter_flags_.trace_flags_ |= provider.get().trace_flags_;
 
             for (const auto& filter : provider.get().filters_) {
-                if (filter.provider_filter_event_id() > 0) {
+                if (filter.provider_filter_event_ids().size() > 0) {
                     //native id existing, set native filters
                     auto& provider_filter_event_ids = provider_flags[provider.get().guid_].provider_filter_event_ids_;
```

Comment:
`provider_flags[provider.get().guid_].provider_filter_event_ids_;` could be `flags.provider_filter_event_ids_`.